### PR TITLE
Add sync with editor command and use different icons for package representation

### PR DESCRIPTION
**Pull request by Flanker32**
1. Add sync command to synchronize/desynchronize dependency viewer selection with folder explorer
2. Split `changeRepresentation` to `changeToFlatPackageView` and `changeToHierarchicalPackageView`, use different icons.
hierarchicalpackage

### CODE REVIEW

File: images/dark/icon-link.svg [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Generator: Adobe Illustrator 22.1.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
+<svg version="1.1" id="图层_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
```
Comment (jdneo) [inline images/dark/icon-link.svg]: `图层_1`?

File: package.json [hunk 2]
```diff
@@ -143,4 +180,4 @@
     "vscode-extension-telemetry-wrapper": "^0.3.8",
     "xml2js": "^0.4.19"
   }
-}
+}
```
Comment (jdneo) [inline package.json]: Always add an empty line at the end of each file

**Comment by jdneo** [comment]
Would be great if gif/screenshot is provided

File: src/views/dependencyExplorer.ts [hunk 3]
```diff
@@ -35,7 +35,7 @@ export class DependencyExplorer {
         });
 
         this._dataProvider.onDidChangeTreeData(() => {
-            if (window.activeTextEditor && Settings.syncWithFolderExplorer()) {
```
Comment (Flanker32) [inline src/views/dependencyExplorer.ts]: Shall we sync with editor when tree data changed (like change the representation of package) if `syncWithFolderExplorer` is false? If we keep this conditon, tree view will not refresh when user set `syncWithFolderExplorer` to true for `setting.js` will call refresh first and then update the dependencyConfig when configuration changed

File: src/settings.ts [hunk 4]
```diff
@@ -13,16 +13,40 @@ export class Settings {
...
             }
-            this._depdendencyConfig = updatedConfig;
+            this._dependencyConfig = updatedConfig;
 
         }));
 
-        const instrumented = instrumentOperation(Commands.VIEW_PACKAGE_CHANGEREPRESENTATION, Settings.changePackageRepresentation);
-        context.subscriptions.push(commands.registerCommand(Commands.VIEW_PACKAGE_CHANGEREPRESENTATION, instrumented));
+        context.subscriptions.push(commands.registerCommand(Commands.VIEW_PACKAGE_SYNCWITHFOLDER,
+            instrumentOperation(Commands.VIEW_PACKAGE_SYNCWITHFOLDER, Settings.syncWithFolderCommand)));
```
Comment (yaohaizh) [inline src/settings.ts]: Could we use only one command for the operation, just dynamically change the icon?

File: package.json [hunk 5]
```diff
@@ -56,6 +56,33 @@
           "dark": "images/dark/icon-hierarchical.svg",
           "light": "images/light/icon-hierarchical.svg"
         }
+      },
+      {
+        "command": "java.view.package.changeToHierarchicalPackageView",
+        "title": "%contributes.commands.java.view.package.changeToHierarchicalPackageView%",
```
Comment (yaohaizh) [inline package.json]: Could we merge these two commands as one? And dynamically change the icon?

File: src/commands.ts [hunk 6]
```diff
@@ -10,7 +10,13 @@ export namespace Commands {
      */
     export const EXECUTE_WORKSPACE_COMMAND = "java.execute.workspaceCommand";
 
-    export const VIEW_PACKAGE_CHANGEREPRESENTATION = "java.view.package.changeRepresentation";
+    export const VIEW_PACKAGE_CHANGETOFLATPACKAGEVIEW = "java.view.package.changeToFlatPackageView";
+
+    export const VIEW_PACKAGE_CHANGETOHIERARCHICALPACKAGEVIEW = "java.view.package.changeToHierarchicalPackageView";
+
+    export const VIEW_PACKAGE_SYNCWITHFOLDER = "java.view.package.syncWithFolderExplorer";
+
```
Comment (andxu) [inline src/commands.ts]: linkWith...

File: src/commands.ts [hunk 7]
```diff
@@ -10,7 +10,13 @@ export namespace Commands {
...
+    export const VIEW_PACKAGE_DESYNCWITHFOLDER = "java.view.package.desyncWithFolderExplorer";
```
Comment (andxu) [inline src/commands.ts]: unlinkXXX

File: src/settings.ts [hunk 8]
```diff
@@ -13,36 +13,59 @@ export class Settings {
                 return;
             }
             const updatedConfig = workspace.getConfiguration("java.dependency");
-            if (updatedConfig.showOutline !== this._depdendencyConfig.showOutline
-                || updatedConfig.packagePresentation !== this._depdendencyConfig.packagePresentation) {
+            if (updatedConfig.showOutline !== this._dependencyConfig.showOutline
+                || updatedConfig.packagePresentation !== this._dependencyConfig.packagePresentation
+                || (updatedConfig.syncWithFolderExplorer !== this._dependencyConfig.syncWithFolderExplorer
+                    && updatedConfig.syncWithFolderExplorer)) {
+                this._dependencyConfig = updatedConfig;
```
Comment (andxu) [inline src/settings.ts]: ` this._dependencyConfig = updatedConfig;` duplicate during if and else

Comment (Flanker32) [reply src/settings.ts]: We will call refresh when we found configuration changed,  and some functions triggered by refresh command will consult configurations, so i call  `this._dependencyConfig = updatedConfig` before execution.

**Pull request merged**